ethereum/spec/src/main/java/tech/pegasys/teku/spec/Spec.java

```diff
@@ -326,26 +328,47 @@ public SignedBeaconBlock deserializeSignedBeaconBlock(final Bytes serializedSign
   }
 
   public SignedBlockContainer deserializeSignedBlockContainer(
-      final Bytes serializedSignedBlockContainer) {
-    final UInt64 slot =
-        BeaconBlockInvariants.extractSignedBlockContainerSlot(serializedSignedBlockContainer);
-    return atSlot(slot)
-        .getSchemaDefinitions()
+      final Bytes serializedSignedBlockContainer, final Optional<String> milestone) {
+
+    final SchemaDefinitions schemaDefinition =
+        getSchemaDefinitionsForMilestone(milestone)
+            .orElseGet(getSchemaDefinitionForSignedBlockSSZ(serializedSignedBlockContainer));
+
+    return schemaDefinition
         .getSignedBlockContainerSchema()
         .sszDeserialize(serializedSignedBlockContainer);
   }
 
   public SignedBlockContainer deserializeSignedBlindedBlockContainer(
-      final Bytes serializedSignedBlindedBlockContainer) {
-    final UInt64 slot =
-        BeaconBlockInvariants.extractSignedBlockContainerSlot(
-            serializedSignedBlindedBlockContainer);
-    return atSlot(slot)
-        .getSchemaDefinitions()
+      final Bytes serializedSignedBlindedBlockContainer, final Optional<String> milestone) {
+
+    final SchemaDefinitions schemaDefinition =
+        getSchemaDefinitionsForMilestone(milestone)
+            .orElseGet(getSchemaDefinitionForSignedBlockSSZ(serializedSignedBlindedBlockContainer));
+
+    return schemaDefinition
         .getSignedBlindedBlockContainerSchema()
         .sszDeserialize(serializedSignedBlindedBlockContainer);
   }
 
+  private Optional<SchemaDefinitions> getSchemaDefinitionsForMilestone(
```

Comment:
maybe can rename this to `getSchemaDefinitionsForEthConsensusVersion(final Optional<String> ethConsensusVersion)` just to clarify it a bit more

Reply:
EthConsensusVersion is a concept of the API layer. In this case, what we are searching is a schema for a milestone so I think my name makes a bit more sense. I'll keep it as is but we can always revisit it later if that's the case.